zebra-consensus/src/block/check.rs

```diff
@@ -184,14 +185,42 @@ pub fn subsidy_is_valid(
         // Note: Canopy activation is at the first halving on mainnet, but not on testnet
         // ZIP-1014 only applies to mainnet, ZIP-214 contains the specific rules for testnet
         // funding stream amount values
-        let funding_streams = zebra_chain::parameters::subsidy::funding_stream_values(
+        let mut funding_streams = zebra_chain::parameters::subsidy::funding_stream_values(
             height,
             network,
             expected_block_subsidy,
         )
         // we always expect a funding stream hashmap response even if empty
         .map_err(|err| BlockError::Other(err.to_string()))?;
 
+        let has_expected_output = |address, expected_amount| {
+            subsidy::funding_streams::filter_outputs_by_address(coinbase, address)
```

Comment:
Potentially add a helper method to the transaction type that finds the outputs for a given address, or move this into the subsidy code.

Reply:
Simplified the logic, removed `filter_outputs_by_address()`, and moved `new_coinbase_script()` up a level to the `subsidy` module in https://github.com/ZcashFoundation/zebra/pull/9603/commits/12cc25d3eb516dd3a31e2fde9bd1a8982bc514d1.